tests/console/validate_lvm_encrypt.pm

```diff
@@ -84,13 +152,75 @@ sub run {
         die "LVM usage stats do not differ!";
     }
 
-    record_info('INFO', 'Check partitions');
-    if ((get_var('NAME') =~ m/separate/) || is_aarch64 && !is_storage_ng) {    # Separate boot partition, not encrypted. aarch64+!storage_ng: see poo#49718
-        assert_script_run q[df | grep -P "^\/dev\/.{3}\d{1}"| grep boot];
+    record_info('LAYOUT', 'Partition layout overview');
+    my $drives = extract_drives(decode_json(script_output qq[lsblk -p -o NAME,TYPE,MOUNTPOINT -J -e 11]));
+    record_info('DRIVES', 'Total = ' . scalar @{$drives});
+    my $parts = extract_partitions_from_disk($drives);
+    record_info('PARTS', 'Total = ' . scalar @{$parts});
+    my $crypts = get_crypt_containers($parts);
+    record_info('CRYPT', 'Total = ' . scalar @{$crypts});
+    my $encrypted_volumes = get_encrypted_volumes($crypts);
+    record_info('VOLUMES',  'Total = ' . scalar @{$encrypted_volumes});
+    record_info($_->{name}, "Mounted in $_->{mountpoint} ($_->{type})") foreach (@{$encrypted_volumes});
```

Comment:
I wonder if we could simply put it in `test_data` structure instead of complex parsing?

Reply:
We can and I would generally recommend that because we already know what kind of partition layout we want to get and we should be aware of how the yast2 partition work. However, I would still keep the code since we need to somehow verify whether we are getting want we requested.

Reply:
Right, we should use this structure in both places, basically.

Reply:
I would suggest so, then it is only about to compare 2 complex perl hash structures, but I guess there will be some CPAN package which can do the job.

Reply:
Yeah, there is `is_deeply` in `Test::More`, but we don't have it available on all the workers yet.

Reply:
Actually, what I wanted to mention, you have done much more than I've expected, as originally we have talked about `cryptlvm` only. So that's good. Have you tried it with openSUSE?

Reply:
not yet, because I did not want to spend much time with updating *main.pm* files and I have not noticed any yaml schedule files for oS. I can try

Reply:
Sure, feel free to delegate it to us, so you can focus on JeOS.

Reply:
oS works fine, but I have used YAML files for schedule. Should I push it with YAML or do you want to update main.pm for o3 ?

Reply:
I would go for yaml, tbh.

Reply:
o3 yaml vrs:

* [opensuse-Tumbleweed-NET-aarch64-Build20190712-cryptlvm@aarch64](http://eris.suse.cz/tests/16439)
* [opensuse-Tumbleweed-DVD-x86_64-Build20190721-cryptlvm@uefi](http://eris.suse.cz/tests/16441)
* [opensuse-Tumbleweed-DVD-x86_64-Build20190721-cryptlvm@64bit](http://eris.suse.cz/tests/16443)